---
title: "Module 6 HW"
author: "Zachary Gonzalez"
date: "2/24/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}

```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}

```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

QUESTION: On the Golub et al. (1999) data, consider the â€œH4/j geneâ€ gene (row 
2972) and the â€œAPS Prostate specific antigenâ€ gene (row 2989). Setup the 
appropriate hypothesis for proving the following claims. Chose and carry out the 
appropriate tests.
(a) The mean â€œH4/j geneâ€ gene expression value in the ALL group is greater than -
0.9. 
(b) The mean â€œH4/j geneâ€ gene expression value in ALL group differs from the 
mean â€œH4/j geneâ€ gene expression value in the AML group.
(c) In the ALL group, the mean expression value for the â€œH4/j geneâ€ gene is lower 
than the mean expression value for the â€œAPS Prostate specific antigenâ€ gene. 
(d) Let pH4j denotes the proportion of patients for whom the â€œH4/j geneâ€ expression 
values is greater than -0.6. We wish to show that pH4j in the ALL group is less than 
0.5.
(e) The proportion pH4j in the ALL group differs from the proportion pH4j in the 
AML group. 

Question 1 - part a 

Hypothesis: H0: p = 0.9, HA: p > 0.9 

```{r}
library(multtest)
data(golub)
```

```{r}
gol.fac <- factor(golub.cl, levels=0:1, labels = c("ALL", "AML"))
t.test(golub[2972,gol.fac=="ALL"],mu=0.9,alternative = "greater")
```
The p value is 1, significantly greater than the p-value of 0.05. We do not have the evidence to reject the the null hypothesis that the mean gene expression of H4/j gene in ALL group is equal to 0.9

Question 1 - part b 

Hypothesis: H0: mux = muy, HA: mux != muy - mux denotes mean gene expression for H4/j ALL and muy denotes mean gene expression for H4/j AML group 

```{r}
t.test(golub[2972,gol.fac=="ALL"],golub[2972,gol.fac=="AML"])
```

the p value is 0.1444, significantly greater than the p value of 0.05 thus we do not have the evidence to reject the null hypothesis that mean gene expression for H4/j ALL is equal to mean gene expression for H4/j AML.  

Question 1 - part c 

Hypothesis: H0: H4/j pALL = APS pALL, HA: H4/j pALL < APS pALL

```{r}
t.test(golub[2972,gol.fac=="ALL"],golub[2989,gol.fac=="ALL"],alternative = "less")
```
the p value is 0.045. This is less than than the value of 0.05. Thus, we can reject the null hypothesis that mean gene expression is the same for the H4/j gene and the APS Prostate antigen gene in the ALL group. 

Question 1 - part d 

Hypothesis: H0: pH4j ALL = 0.5, HA: pH4j ALL < 0.5

use binom.test

```{r}
pH4j<-sum(golub[2972,]>-0.6)
n=length(golub[2972,])
binom.test(x=pH4j,n=n,p=0.5,alternative = "less")
```
The p-value is greater than the p value of 0.05. We cannot reject the null hypothesis that pH4j in the ALL group is equal to 0.5. 

Question 1 - part e - use prop.test

Hypothesis: H0: pH4j ALL = pH4j AML, HA: pH4j ALL != pH4j AML

```{r}
pH4jALL<-sum(golub[2972,gol.fac=="ALL"]>-0.6)
pH4jAML<-sum(golub[2972,gol.fac=="AML"]>-0.6)
n=length(golub[2972,gol.fac=="ALL"])
n2=length(golub[2972,gol.fac=="AML"])
prop.test(x=c(pH4jALL,pH4jAML),n=c(n,n2),alternative = "two.sided")
```

The p-value is 0.101, greater than the p-value of 0.05. We do not have the evidence to reject the null hypothesis that the proportion of H4j gene expression in the ALL group is equal to the proportion of H4j gene expression in the AML group. 

QUESTION: Suppose that the probability to reject a biological hypothesis by the 
results of a certain experiment is 0.05. This experiment is repeated 2000 times.
(a) How many rejections do you expect?
(b) What is the probability of less than 90 rejections?

Question 2 - part a 

```{r}
p = 0.05
n = 2000
num_of_rejections = n * p
num_of_rejections
```
We would expect 100 rejections 

Question 2 - part b 

```{r}
binom.test(x=90,n=2000,p=0.05,alternative = "less")
```

QUESTION: For testing H0: Î¼=3 versus HA: Î¼>3, we considers a new Î±=0.1 level test which 
rejects when ð‘¡ð‘œð‘ð‘  = ð‘‹â€•3
ð‘ / ð‘› falls between ð‘¡0.3,ð‘› â€•1 and ð‘¡0.4,ð‘› â€•1. 
(a) Use a Monte Carlo simulation to estimate the Type I error rate of this test 
when n=20. Do 10,000 simulation runs of data sets from the 
ð‘(ðœ‡ = 3,ðœŽ= 4). Please show the R script for the simulation, and the R 
outputs for running the script. Provide your numerical estimate for the Type 
I error rate. Is this test valid (that is, is its Type I error rate same as the 
nominal Î±=0.1 level)? 
(b) Should we use this new test in practice? Why or why not? 

Question 3 - part a 
x.mean=apply(x.sim,1,mean)
x.sd=
tstat=(x.mean-3)/x.sd*sqrt(20)
rej=tstat>qt(0.3,df=19)&
mean(rej)

```{r}
x.sim<-matrix(rnorm(10000*20, mean=3,sd=4),ncol=20)
x.mean<-apply(x.sim,1,mean)
x.sd<-apply(x.sim,2,sd)
tstat<-(x.mean-3)/x.sd*sqrt(20)
rejec<-tstat>qt(0.3,df=19)&tstat<qt(0.4,df=19)
mean(rejec)
```

Question 3 - part b

The type 1 error rate provided by this test is very close to the nominal value of 0.1 so yes, this test should be used in practice. 

Question 4 - compare all gene expression values in the entire golub data set


```{r}
gol.fac<-factor(golub.cl,levels=0:1, labels=c("ALL","AML"))
p.valuesGolub<-apply(golub,1,function(x) t.test(x~gol.fac))$p.value
sum(p.valuesGolub<0.05)
```

QUESTION: On the Golub et al. (1999) data set, do Welch two-sample t-tests to compare 
every geneâ€™s expression values in ALL group versus in AML group. 
(a) Use Bonferroni and FDR adjustments both at 0.05 level. How many genes 
are differentially expressed according to these two criteria? 
(b)Find the gene names for the top three strongest differentially expressed 
genes (i.e., minimum p-values). Hint: the gene names are stored in 
golub.gnames.

Question 4 - part a - bonferonni and fdr adjustments  

```{r}
p.bonGolub<-p.adjust(p=p.valuesGolub, method="bonferroni")
p.fdrGolub<-p.adjust(p=p.valuesGolub, method="fdr")
sum(p.bonGolub<0.05)
sum(p.fdrGolub<0.05)
```
1827 significantly expressed genes according to the bonferonni adjustment 
2512 significantly expressed genes according to the fdr adjustment 

Question 4 - part b 

```{r}
orderGenes<-order(p.valuesGolub, decreasing = FALSE)
print(golub.gnames[orderGenes[1:3],2])
```

